### PR TITLE
Update to 1.12.2

### PR DESCRIPTION
**Pull request by cleverpanda**
Just ported it directly for you. Been trying to contact you on various outlets so hit me up when you see this please :)

### CODE REVIEW

**Comment by rockhymas** [comment]
Thank you so much for tackling this. I've been trying to wrap up some other projects so I could get back to it. You're awesome. Now for some feedback :)

File: CREDITS-fml.txt [hunk 1]
```diff
@@ -0,0 +1,28 @@
+This is Forge Mod Loader.
```
Comment (rockhymas) [inline CREDITS-fml.txt:1]: Remove this file from the PR

File: LICENSE-new.txt [hunk 2]
```diff
@@ -0,0 +1,483 @@
+Minecraft Forge is licensed under the terms of the LGPL 2.1 found
```
Comment (rockhymas) [inline LICENSE-new.txt:1]: Remove this file from the PR

File: MinecraftForge-Credits.txt [hunk 3]
```diff
@@ -0,0 +1,26 @@
+* Eloraam *
```
Comment (rockhymas) [inline MinecraftForge-Credits.txt:1]: Remove this file from the PR

File: Paulscode IBXM Library License.txt [hunk 4]
```diff
@@ -0,0 +1,10 @@
+IBXM is copyright (c) 2007, Martin Cameron, and is licensed under the BSD License.
```
Comment (rockhymas) [inline Paulscode IBXM Library License.txt:1]: Remove this file from the PR

File: Paulscode SoundSystem CodecIBXM License.txt [hunk 5]
```diff
@@ -0,0 +1,40 @@
+SoundSystem CodecIBXM Class License:
```
Comment (rockhymas) [inline Paulscode SoundSystem CodecIBXM License.txt:1]: Remove this file from the PR

File: README.txt [hunk 6]
```diff
@@ -0,0 +1,59 @@
+-------------------------------------------
```
Comment (rockhymas) [inline README.txt:1]: Remove this file from the PR

File: forge-1.12.2-14.23.0.2508-changelog.txt [hunk 7]
```diff
@@ -0,0 +1,13024 @@
+Changelog:
```
Comment (rockhymas) [inline forge-1.12.2-14.23.0.2508-changelog.txt:1]: Remove this file from the PR

File: gradle.properties [hunk 8]
```diff
@@ -0,0 +1,3 @@
+# Sets default memory used for gradle commands. Can be overridden by user or command line properties.
```
Comment (rockhymas) [inline gradle.properties:1]: Hm, I've never needed this. I believe we can remove it.

File: src/main/java/com/gibraltar/iberia/challenge/ArmorChallenge.java [hunk 9]
```diff
@@ -341,10 +314,10 @@ public void onEntityInteractSpecific(PlayerInteractEvent.EntityInteractSpecific
 		else {
 			boolean isSmall = armorStand.isSmall();
 			Vec3d vec = event.getLocalPos();
-			double d4 = isSmall ? vec.yCoord * 2.0D : vec.yCoord;
+			double d4 = isSmall ? vec.y * 2.0D : vec.y;
 			ItemStack stack = event.getItemStack();
 
-			if (stack == null || !(stack.getItem() instanceof ItemElytra) || event.getHand() != EnumHand.MAIN_HAND) {
+			if (stack.isEmpty() || !(stack.getItem() instanceof ItemElytra) || event.getHand() != EnumHand.MAIN_HAND) {
```
Comment (rockhymas) [inline src/main/java/com/gibraltar/iberia/challenge/ArmorChallenge.java:320]: How certain are you that `event.getItemStack()` will never return null?

File: src/main/java/com/gibraltar/iberia/challenge/ArmorChallenge.java [hunk 10]
```diff
@@ -440,7 +413,7 @@ public void onEnchantmentLevelSet(EnchantmentLevelSetEvent event) {
 
 	@SubscribeEvent
 	public void onRightClickBlock(PlayerInteractEvent.RightClickBlock event) {
-		boolean itemIsBlock = event.getItemStack() != null && event.getItemStack().getItem() instanceof ItemBlock;
+		boolean itemIsBlock = !event.getItemStack().isEmpty() && event.getItemStack().getItem() instanceof ItemBlock;
```
Comment (rockhymas) [inline src/main/java/com/gibraltar/iberia/challenge/ArmorChallenge.java:416]: And here.

File: src/main/java/com/gibraltar/iberia/challenge/HealingChallenge.java [hunk 11]
```diff
@@ -70,6 +66,10 @@ public void onPlayerWakeUp(PlayerWakeUpEvent event) {
                     case EASY:
                         player.heal(healAmountEasy);
                         break;
+				case PEACEFUL:
```
Comment (rockhymas) [inline src/main/java/com/gibraltar/iberia/challenge/HealingChallenge.java:69]: Indentation looks like we've got a tabs vs spaces issue here.

File: src/main/java/com/gibraltar/iberia/challenge/SpawnChallenge.java [hunk 12]
```diff
@@ -118,6 +113,10 @@ private int getDistanceToNextSpawn(World world) {
             case EASY:
                 distanceToNewSpawn = distanceToNewSpawnEasy;
                 break;
+		case PEACEFUL:
+			break;
```
Comment (rockhymas) [inline src/main/java/com/gibraltar/iberia/challenge/SpawnChallenge.java:117]: More tabs vs spaces

File: src/main/resources/pack.mcmeta [hunk 13]
```diff
@@ -0,0 +1,7 @@
+{
+    "pack": {
+        "description": "examplemod resources",
```
Comment (rockhymas) [inline src/main/resources/pack.mcmeta:3]: Let's change this to "Iberia resources"

File: src/main/java/com/gibraltar/iberia/challenge/ArmorChallenge.java (hunk 9 above)
Comment (cleverpanda) [reply src/main/java/com/gibraltar/iberia/challenge/ArmorChallenge.java]: Itemstacks can NEVER be null anymore.

Comment (cleverpanda) [reply src/main/java/com/gibraltar/iberia/challenge/ArmorChallenge.java:320]: Itemstacks cant be null now.

and

```public ItemStack getItemStackFromSlot(EntityEquipmentSlot slotIn)
    {
        if (slotIn == EntityEquipmentSlot.MAINHAND)
        {
            return this.inventory.getCurrentItem();
        }
        else if (slotIn == EntityEquipmentSlot.OFFHAND)
        {
            return this.inventory.offHandInventory.get(0);
        }
        else
        {
            return slotIn.getSlotType() == EntityEquipmentSlot.Type.ARMOR ? (ItemStack)this.inventory.armorInventory.get(slotIn.getIndex()) : ItemStack.EMPTY;
        }
    }```

Comment (rockhymas) [reply src/main/java/com/gibraltar/iberia/challenge/ArmorChallenge.java:320]: Nice!

**Pull request closed without merging**